src/sentry/profiles/task.py

```diff
@@ -43,6 +48,18 @@ def process_profile(profile: MutableMapping[str, Any], **kwargs: Any) -> None:
         json.dumps(profile),
     )
 
+    track_outcome(
+        org_id=project.organization_id,
+        project_id=project.id,
+        key_id=None,
+        outcome=Outcome.ACCEPTED,
+        reason=None,
+        timestamp=datetime.utcnow().replace(tzinfo=UTC),
+        event_id=profile["transaction_id"],
```

Comment:
Relay serializes the `transaction_id` as `Uuid`, which includes hyphens. In outcomes, so far we only produce the event ID without hyphens in simple representation. Snuba [can handle both](https://github.com/getsentry/snuba/blob/5d84bb73435847e2e405061666b85fe3cfce3a2c/snuba/datasets/outcomes_processor.py#L80)  and will re-hyphenate it anyway, but if you want to be super consistent, you should ensure to use the `EventId` type in Relay for this field.

Reply:
@jan-auer good point. Thanks for bringing that up.

Either that or I can define a custom serialization for `transaction_id` with: `#[serde(serialize_with = "transaction_id_serialize")]`

But probably using the `EventId` as you suggest is the safest as I don't have to add custom serialization for transaction_id in all the structs: cocoa, android (and rust in the future)